src/Donations/resources/admin/components/DonationDetailsPage/Tabs/Records/General/AssociatedDonor/useDonorAsyncSelect.ts

```diff
@@ -0,0 +1,102 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useEntityRecord } from '@wordpress/core-data';
+import apiFetch from '@wordpress/api-fetch';
+import { Donor } from '@givewp/donors/admin/components/types';
+import {
+    DonorOption,
+    UseDonorAsyncSelectReturn
+} from './types';
+import {
+    formatDonorOption,
+    formatDonorOptions,
+    processOptionsForMenu,
+    createDonorQueryParams
+} from './utils';
+import { getDonationOptionsWindowData } from '@givewp/donations/utils';
+
+const DONORS_PER_PAGE = 30;
+
+/**
+ * Custom hook for handling async donor selection with pagination and search
+ *
+ * @unreleased
+ */
+export function useDonorAsyncSelect(selectedDonorId: number | null): UseDonorAsyncSelectReturn {
+    const { mode } = getDonationOptionsWindowData();
+    const [page, setPage] = useState(0);
+    const [selectedOption, setSelectedOption] = useState<DonorOption | null>(null);
+    const [error, setError] = useState<Error | null>(null);
+
+    // Load current donor details if we have a selected donor ID
+    const {
+        record: currentDonor,
+        hasResolved: hasResolvedDonor,
+    } = useEntityRecord<Donor>('givewp', 'donor', selectedDonorId);
+
+    // Update selected option when current donor changes
+    useEffect(() => {
+        if (hasResolvedDonor && currentDonor && selectedDonorId) {
+            setSelectedOption(formatDonorOption(currentDonor));
+        } else if (!selectedDonorId) {
+            setSelectedOption(null);
+        }
+    }, [currentDonor, hasResolvedDonor, selectedDonorId]);
+
+    // Load options function for AsyncPaginate
+    const loadOptions = useCallback(async (searchInput: string) => {
+        const currentPage = searchInput !== '' ? 1 : page + 1;
```

Comment:
I think we can simplify it this way: 

`const currentPage = searchInput ? 1 : page + 1;`

Reply:
Resolved on 30f962d939fbbac7effe624e6f4ea49bc9addd5b